### PR TITLE
refactor (build): PRF-7071 remove dead code

### PR DESCRIPTION
**Pull request by ParnikaGupta9**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-7071 Remove the depreciated functionalities- Level four and three](https://app.clickup.com/t/9002019994/PRF-7071)

File: src/card_operations/card_pair.c [hunk 1]
```diff
@@ -163,6 +167,50 @@ static void init_and_pair_card(card_operation_data_t *card_data,
...
+    atcab_init(atecc_data.cfg_atecc608a_iface);
+    atecc_data.status = atcab_sign(slot_3_nfc_pair_key, hash, sign);
+  } while (atecc_data.status != ATCA_SUCCESS && --atecc_data.retries);
+  if (usb_irq_enable_on_entry == true)
+    NVIC_EnableIRQ(OTG_FS_IRQn);
+
+  return atecc_data.status;
+}
+
+static uint8_t atecc_nfc_ecdh(const uint8_t *pub_key, uint8_t *shared_secret) {
```
Comment (ujjwal-cyph) [inline src/card_operations/card_pair.c:190]: add prototypes of these function under the `"STATIC FUNCTION PROTOTYPES"` section of this file.

File: src/card_operations/card_internal.h [hunk 2]
```diff
@@ -13,17 +13,44 @@
...
+ * @brief
+ * @details
+ *
+ * @see
+ * @since v1.0.0
+ *
+ * @note
+ */
+typedef struct NFC_connection_data {
+  uint8_t desktop_control;
```
[the remark this answers is not in the document]
Comment (VeronicaGupta) [reply src/card_operations/card_internal.h:38]: ok

File: src/card_operations/card_pair.c (hunk 1 above)
Comment (VeronicaGupta) [reply src/card_operations/card_pair.c:190]: ok

**Review by VeronicaGupta — APPROVED** [review]

**Pull request closed without merging**